File: dj-client/src/apis/AuthApis/AuthApis.ts
import TokenModel from "@/entities/AuthEntities/TokenModel";
import { HTTP } from "../HTTPs";
class AuthApis {
  getLogin(requestLogin: object): Promise<any> {
    return new Promise<any>((resolve: any, reject: any) => {
      HTTP.post(`/login/getlogin`, requestLogin)
        .then((response) => {
          resolve(response.data);
        })
        .catch((error) => {
          reject(error);
        });
    });
  }

  refreshToken(tokenModel: TokenModel): Promise<any> {
    return new Promise<any>((resolve: any, reject: any) => {
      HTTP.post(`/login/refresh`, tokenModel)
        .then((response) => {
          resolve(response.data);
        })
        .catch((error) => {
          reject(error);
        });
    });
  }
  signIn(newAccount: object): Promise<any> {
    return new Promise<any>((resolve: any, reject: any) => {
      HTTP.post(`/login/signin`, newAccount)
        .then((response) => {
          resolve(response.data);
        })
        .catch((error) => {
          reject(error);
        });
    });
  }
  confirmAccount(code: string): Promise<any> {
    return new Promise<any>((resolve: any, reject: any) => {
      HTTP.get(`/login/confirmemail?code=` + code)
        .then((response) => {
          resolve(response.data);
        })
        .catch((error) => {
          reject(error);
        });
    });
  }
  forgetPass(forgetRequest: string): Promise<any> {
    return new Promise<any>((resolve: any, reject: any) => {
      HTTP.post(`/login/forgetpass`, forgetRequest)
        .then((response) => {
          resolve(response.data);
        })
        .catch((error) => {
          reject(error);
        });
    });
  }
  confirmCode(code: string): Promise<any> {
    return new Promise<any>((resolve: any, reject: any) => {
      HTTP.get(`login/confirmcodeforgetpass?code=${code}`)
        .then((response) => {
          resolve(response.data);
        })
        .catch((error) => {
          reject(error);
        });
    });
  }
  confirmPass(request: object): Promise<any> {
    return new Promise<any>((resolve: any, reject: any) => {
      HTTP.post(`login/confirmpass`, request)
        .then((response) => {
          resolve(response.data);
        })
        .catch((error) => {
          reject(error);
        });
    });
  }
}
export default new AuthApis();

File: dj-endpoint/Controllers/AuthApis/AuthApis.cs

using dj_actionlayer.Business.Auth;
using dj_webdesigncore.AuthModel;
using dj_webdesigncore.Business.Auth;
using dj_webdesigncore.Request.Account;
using Microsoft.AspNetCore.Mvc;

namespace dj_endpoint.Controllers.AuthApis
{
    [Route("/login")]
    public class AuthApis : BaseApi
    {
        private readonly IAuthBusiness authBusiness;

        public AuthApis()
        {
            this.authBusiness = new AuthBusiness();
        }
        [HttpPost("getlogin")]
        public async Task<IActionResult> Login(RequestLogin request)
        {
            return Ok(await authBusiness.Login(request));
        }
        [HttpPost("refresh")]
        public async Task<IActionResult> RefreshToken(TokenModel tokenModel)
        {
            return Ok(await authBusiness.RenewToken(tokenModel));
        }
        [HttpPost("signin")]
        public async Task<IActionResult> SignIn(SignInAccount newAccount)
        {
            return Ok(await authBusiness.SignIn(newAccount));
        }
        [HttpGet("confirmemail")]
        public async Task<IActionResult> ConfirmEmail(string code)
        {
            return Ok(await authBusiness.ConfirmAccount(code));
        }
        [HttpPost("forgetpass")]
        public async Task<IActionResult> ForgetPass(ForgetPassRequest forgetPassRequest)
        {
            return Ok(await authBusiness.ForgetPass(forgetPassRequest));
        }
        [HttpGet("confirmcodeforgetpass")]
        public async Task<IActionResult> ConfirmCodeForgetPass(string code)
        {
            return Ok(await authBusiness.ConfirmCodeForgetPass(code));
        }
        [HttpPost("confirmpass")]
        public async Task<IActionResult> ConfirmNewPass(ConfirmNewPass confirmNewPass)
        {
            return Ok(await authBusiness.ConfirmNewPass(confirmNewPass));
        }
    }
}
